### ISSUE TITLE
Remove twisted.mail.protocols.DomainSMTP and DomainESMTP

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8772]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8772|
|Type|task|
|Created|2016-08-19 13:34:08Z|
|Branch|https://github.com/twisted/twisted/tree/8772-tmprotocolsdomaindelivery|

Deprecated since 2003, and they aren't really great.

<details><summary>Searchable metadata</summary>

```
trac-id__8772 8772
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8772_tmprotocolsdomaindelivery 8772-tmprotocolsdomaindelivery
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1471613648852426 1471613648852426
changetime__1471625605871949 1471625605871949
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Please review. Includes *bonus tests*, to patch over some lines of coverage that the old code included.

PR: https://github.com/twisted/twisted/pull/508

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

great cleanup. Thanks!

please merge!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 45d3b7ee43c405eaf815797dc9f46ebdfee186af
```
#!CommitTicketReference repository="" revision="45d3b7ee43c405eaf815797dc9f46ebdfee186af"
Merge 8772-tmprotocolsdomaindelivery: Remove twisted.mail.protocols.Domain(E)SMTP

Author: hawkowl
Reviewer: adiroiban
Fixes: #8772
```
